### ISSUE TITLE
Some plugins are not being cleaned up when exiting Pupil

### ISSUE THREAD

**Issue by toshikurauchi** [pupil-labs/pupil#91]
Because of that surfaces sometimes are not saved on Marker_Detector plugin, for example.

It happens because of the way plugins are removed from the list in clean() from Plugin_List.
Removing elements inside a for loop may cause problems in some cases:

```
a = range(10)
for x in a:
     a.remove(x)
print a # [1, 3, 5, 7, 9]
```

**Comment by toshikurauchi** [comment]
Fixed here: #92

**Comment by mkassner** [comment]
Thanks for the catch and the fix! I merged it.
